File: sort/C++/sort.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

#define N (1 << 22)

int main(int argc, char* argv[])
{
    bool print_out = false;
    if(argc >= 2)
        print_out = true;
    vector<int> v(N);
    int seed = 3;
    for(int i = 0; i < N; i++)
    {
        v[i] = seed;
        seed *= 27487;
        seed %= 30491;
    }
    sort(v.begin(), v.end());
    if(print_out)
        cout << v[279121] << endl;
    return 0;
}

File: sort/PHP/sort.php
<?php

ini_set('memory_limit', '-1');

$seed = 3;
$array = array();

for ($i = 0; $i < 4194304; $i++) {
    $array[$i] = $seed;
    $seed = $seed * 27487;
    $seed = $seed % 30491;
}

sort($array);

if (count($argv) > 1)
    echo $array[279121];
